### ISSUE TITLE
Disallow noop `setupOnerror` in before/beforeEach

### ISSUE THREAD

**Issue by v-korshun** [ember-cli/eslint-plugin-ember#918]
Right now nothing prevents engineers to setup noop error handler in before/beforeEach which will swallow all rejection and errors in the module. We should not allow to use noop setupOnerror in before/beforeEach since it could mask errors/rejections in tests unintentionally.

**Comment by bmish** [comment]
Can you a good example and a bad example to help explain what you mean?

**Comment by v-korshun** [comment]
This rule forbids the following:

```js

import { setupOnerror } from '@ember/test-helpers';
import { module } from 'qunit';

module('foo', function(hooks) {
  hooks.beforeEach(function() {
    setupOnerror(() => {});
  });
});

```
Or
```

import { setupOnerror } from '@ember/test-helpers';
import { module } from 'qunit';

module('foo', function(hooks) {
  hooks.before(function() {
    setupOnerror(() => {});
  });
});

```


And suggests using the following instead:

```
import { setupOnerror } from '@ember/test-helpers';
import { module, test } from 'qunit';

module('foo', function(hooks) {
  test('something', function() {
    setupOnerror((error) => {
      assert.equal(error.message, 'test', 'Should have message');
    });
  })
});
```

**Comment by rwjblue** [comment]
FWIW, I am in favor of adding a rule for this...
